### PR TITLE
Validation spec

### PR DESCRIPTION
**Pull request by EvansSnave**
Add the following validations:
For the User model:
Name must not be blank.
PostsCounter must be an integer greater than or equal to zero.
For the Post model:
Title must not be blank.
Title must not exceed 250 characters.
CommentsCounter must be an integer greater than or equal to zero.
LikesCounter must be an integer greater than or equal to zero.
Add unit specs for all of your models' methods and validations.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :posts, foreign_key: :author_id
+  has_many :comments, foreign_key: :author_id
+  has_many :likes
+
+  def three_recent_posts
+    posts.order(created_at: :desc).limit(3)
+  end
+end
```
Comment (V-Blaze) [inline app/models/user.rb]: - You have done great so far :+1: But there are a few requirements you need to meet before moving to the next milestone. :+1: :100: 

- - Requirement Screenshot:

Selection_054

- [ ] According to the project requirement, you need to add validations to the user model to make sure that a name is always present t when a user is created and also that PostsCounter must be an integer greater than or equal to zero. Currently, you haven't implemented this. I suggest you do as it is part of the requirement, :+1: :+1: :100: Here is a snippet of what the first validation could look like:
```ruby
validates :name, presence: true
```

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  after_save :update_posts_counter!
```
Comment (V-Blaze) [inline app/models/post.rb:6]: - - Screenshot reference:

Selection_055

- [ ] Just as I pointed out above on the `Users model`, the post counter should have some validation according to the project requirement as i have shown in the screenshot above too. :+1: :+1: I suggest you look at the requirements and add all the validation to the post model. Here is a snippet of the `CommentsCounter must be an integer greater than or equal to zero.` validation could look like. :+1: 

````ruby
validates :comments_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
```

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    let(:user) { create(:user, name: 'John', posts_counter: 2) }
+
+    it 'returns the three most recent posts' do
+      old_post = create(:post, author: user, created_at: 3.days.ago)
+      recent_posts = create_list(:post, 3, author: user)
+
+      expect(user.three_recent_posts).to eq(recent_posts.reverse)
+      expect(user.three_recent_posts).not_to include(old_post)
+    end
+  end
```
Comment (V-Blaze) [inline spec/models/user_spec.rb:29]: - [ ] You have done great writing the model specs and covering all the validations specs and custom methods, but looking at the screenshot below, you can see that only `3 tests` are passing the rest of the tests are failing. I'll suggest you fix these as all the tests are meant to pass. :+1: :smile: :wave: 

- - Screenshot:
Screenshot from 2023-12-03 00-12-33

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @, 

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

almost-there


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- Wrote tests for custom methods✔
- SOme tests are passing👋 
-  Correct git flow✔

 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Strangeal — APPROVED** [review]
Hi @EvansSnave, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: Congratulations! 🎉 




## Highlights
- Added validation to models ✅ 
- Fixed failing test ✅ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**